docs/DataTypes.md

```diff
@@ -0,0 +1,40 @@
+# Data Types
+
+DataHelix currently recognises three distinct data types. Keeping this set small is a deliberate goal; it would be possible to have types like _FirstName_ or _Postcode_, but instead these are considered specialisations of the _String_ type, so they can be constrained by the normal string operators (e.g. a user could generate all first names shorter than 10 characters, starting with a vowel).
+
+## Numeric
+
+In principle, any real number. In practice, any number that can be represented in a Java [BigDecimal](https://docs.oracle.com/javase/7/docs/api/java/math/BigDecimal.html) object.
```

Comment:
Who is the target audience for this documentation? There's the potential of a usability issue if our users have to understand concepts such as big decimal in order to follow what our product does.

Reply:
They should be fairly technical people. We could replicate the BigDecimal docs here instead, but an exhaustive explanation of what numbers can be represented would involve a lot of text.